src/renderers/dom/client/eventPlugins/ChangeEventPlugin.js

```diff
@@ -45,6 +47,35 @@ var eventTypes = {
   },
 };
 
+function getInstIfValueChanged(targetInst, target) {
+  var value = getInputValue(target);
+  var lastValue = target[LAST_VALUE_KEY];
+
+  if (!target.hasOwnProperty(LAST_VALUE_KEY) || value !== lastValue) {
+    target[LAST_VALUE_KEY] = value;
```

Comment:
We can't really do this, updating the value of inputs via script is allowed for uncontrolled inputs and these updates would not update `[LAST_VALUE_KEY]`. Also, as far as I can tell, this implementation does not update `[LAST_VALUE_KEY]` when the value an input is updated for controlled inputs either, but that is fixable (although slightly hacky perhaps).